lib/encoding.neon

```diff
@@ -0,0 +1,159 @@
+%|
+ |  File: encoding
+ |
+ |  Functions for encoding and decoding ASCII or Binary data in various formats.
+ |%
+IMPORT bitwise
+
+EXPORT base64Encode
+EXPORT base64Decode
+
+TYPE Mode IS ENUM
+    BYTE1
+    BYTE2
+    BYTE3
+    BYTE4
+END ENUM
+
+CONSTANT Base64EncodeTable: String := "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
+
+CONSTANT Base64DecodeTable: Array<Number> := [
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0x00 - 0x0f
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0x10 - 0x1f
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,  62, 255, 255, 255,  63,    % 0x20 - 0x2f
+     52,  53,  54,  55,  56,  57,  58,  59,  60,  61, 255, 255, 255,  64, 255, 255,    % 0x30 - 0x3f
+    255,   0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,  14,    % 0x40 - 0x4f
+     15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25, 255, 255, 255, 255, 255,    % 0x50 - 0x5f
+    255,  26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,  40,    % 0x60 - 0x6f
+     41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51, 255, 255, 255, 255, 255,    % 0x70 - 0x7f
+    % This table simply ignores these values.
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0x80 - 0x8f
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0x90 - 0x9f
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0xa0 - 0xaf
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0xb0 - 0xbf
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0xc0 - 0xcf
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0xd0 - 0xdf
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0xe0 - 0xef
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0xf0 - 0xff
+]
+
+%|
+ |  Function: base64Encode
+ |
+ |  Encodes the provided string into Base64 encoded data.
+ |
+ |  Parameters:
+ |      src - The source data to encode.
+ |
+ |%
+FUNCTION base64Encode(src: Bytes): String
+    VAR i: Number := 0
+    VAR encoded: String := ""
+    VAR char1, char2: Number
+    VAR status: Mode := Mode.BYTE1
+    
+    LET source: Array<Number> := src.toArray()
+
+    WHILE i < src.size() DO
```

Comment:
Consider using a `FOR` loop instead of `WHILE` here. You already know ahead of time exactly how many iterations you want to run, and a `FOR` loop is appropriate for that. Also, using a `FOR` loop avoids reevaluating `src.size()` every time.